algebra/src/main/java/org/almostrealism/collect/computations/CollectionPermute.java

```diff
@@ -38,28 +132,99 @@ public CollectionPermute(Producer<?> collection, int... order) {
 		this.order = order;
 	}
 
+	/**
+	 * Returns the memory length required for this computation.
+	 * For CollectionPermute, this is always 1 since it performs a direct index mapping
```

Comment:
This is not the reason - it returns 1 because each kernel thread of the computation operates on only a single element of the collection.